### PR TITLE
Add Elastic Agent in the Jenkinsfile

### PR DESCRIPTION
**Pull request by ph**
## What does this PR do?

With the recent moves to Jenkins pipeline the Elastic Agent project was
not currently tested.

## Why is it important?

To run the tests for the project.

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Relates to https://github.com/elastic/beats/issues/17411

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -55,6 +55,48 @@ pipeline {
     stage('Build and Test'){
       failFast false
       parallel {
+        stage('Elastic Agent x-pack'){
+          agent { label 'ubuntu && immutable' }
+          options { skipDefaultCheckout() }
+          when {
+            beforeAgent true
+            expression {
+              return env.BUILD_ELATIC_AGENT_XPACK != "false"
```
Comment (michalpristas) [inline Jenkinsfile:64]: type ELATIC => ELASTIC

File: Jenkinsfile [hunk 2]
```diff
@@ -55,6 +55,48 @@ pipeline {
...
+          }
+        }
+
+        stage('Elastic Agent Mac OS X'){
+          agent { label 'macosx' }
+          options { skipDefaultCheckout() }
+          when {
+            beforeAgent true
+            expression {
+              return env.BUILD_FILEBEAT != "false" && params.macosTest
```
Comment (michalpristas) [inline Jenkinsfile:92]: FILEBEAT?

**Comment by ph** [comment]
@michalpristas I've created this issue https://github.com/elastic/beats/issues/17449, we don't our logic right when we deal with joining path for the install hooks. I've disabled the test here.

File: x-pack/elastic-agent/pkg/agent/transpiler/steps_test.go [hunk 3]
```diff
@@ -39,16 +41,18 @@ func TestIsSubpath(t *testing.T) {
...
-			{"/", "a", "\\a", true},
-			{"/a", "b", "\\a\\b", true},
-			{"/a", "b/c", "\\a\\b\\c", true},
-			{"/a/b", "/a/c", "\\a\\c", false},
-			{"/a/b", "/a/b/../c", "\\a\\c", false},
-			{"/a/b", "../c", "\\a\\c", false},
-			{"/a", "/a/b/c", "\\a\\b\\c", true},
-			{"/a", "/A/b/c", "\\a\\b\\c", true},
-		},
+		// (Windows issue) See issue:
```
Comment (michalpristas) [inline x-pack/elastic-agent/pkg/agent/transpiler/steps_test.go:44]: 👍

**Comment by michalpristas** [comment]
@ph issue with windows is fixed here #17461

**Comment by ph** [comment]
Jenkins test this please

**Comment by ph** [comment]
jenkins run the tests please

**Comment by ph** [comment]
I am going to merge this as is, and fallback on travis for everything mac related.

**Comment by ph** [comment]
backported in https://github.com/elastic/beats/pull/17599

**Pull request merged**